tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -7416,6 +7418,105 @@ uint32_t SPIRVEmitter::processIntrinsicModf(const CallExpr *callExpr) {
   return 0;
 }
 
+uint32_t SPIRVEmitter::processIntrinsicMad(const CallExpr *callExpr) {
+  // Signature is: ret mad(a,b,c)
+  // All of the above must be a scalar, vector, or matrix with the same
+  // component types. Component types can be float or int.
+  // The return value is equal to  "a * b + c"
+
+  // In the case of float arguments, we can use the GLSL extended instruction
```

Comment:
It's strange to see that Fma with `NoContraction` means actually to **contract** instead of not contract. But that is what is spec'd in GLSL extended instruction set, which is in stark contrast to SPIR-V spec, which says `Apply to an arithmetic instruction to indicate the operation cannot be combined with another instruction to form a single operation.` Although one can argue that `NoContraction` is applied **internally** to Fma itself, so Fma is not merged with **other** instructions, but yet, still very strange use (to the point of almost feel like abuse) of `NoContraction`. Could you make sure that there is no spec issue here?